spiped/main.c

```diff
@@ -274,50 +285,69 @@ main(int argc, char * argv[])
 			warnp_syslog(1);
 	}
 
+	/* Resolve bind address. */
+	if (opt_b) {
+		if ((bind_addr = sock_addr_ensure_port(opt_b))  == NULL) {
+			warnp("Failed to get bind address");
+			goto err2;
+		}
+		while ((sas_b = sock_resolve(bind_addr)) == NULL) {
```

Comment:
Thanks!  If you rebase onto master, you'll find a new `sock_resolve_one()`, which can simplify this code.

I'm also going to submit a further modification to `sock_resolve_one()`, which would include the `sock_addr_ensure_port()` functionality in it.  I don't know if that proposal will be accepted, though.  I'll get back to you when I have news.

Reply:
But there is no `sock_addr_ensure_port()` in  master...

Reply:
Thanks, and sorry for the delay.  There's now a `sock_resolve_one()` in master, which will take care of calling `sock_addr_ensure_port`.